src/test/java/com/aparapi/runtime/ArrayTest.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2016 - 2017 Syncleus, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.aparapi.runtime;
+
+import static org.junit.Assert.assertArrayEquals;
+
+import com.aparapi.Kernel;
+import java.util.Arrays;
+import org.junit.Test;
+
+
+public class ArrayTest {
+    @Test
+    public void test() {
+        VectorKernel b = new VectorKernel();
+        b.test();
+    }
+
+    public static class VectorKernel extends Kernel {
+        static final int size = 32;
+
+        static int[][] target = new int[size][size];
+
+        public VectorKernel() {
+            for (int i = 0; i < size; ++i) {
+                int[] ints = new int[size];
+                Arrays.fill(target[i], 99);
+
+                target[i] = ints;
+            }
+        }
+
+        private static void fillArray(int[] ints_$private$, int id) {
+            for (int i = 0; i < size; i++) {
+                ints_$private$[i] = i + id;
```

Comment:
whats the deal with the "$" symbols here? Unless im missing something this breaks java standard those symbols should never be used in variable names. Also java defines variable names as camel case, they never use underscores (except for constants which are all caps, but that doesnt apply here).

Reply:
Oops ignore this comment. I didnt notice this was being called from the kernel, the $ is the aparapi specific syntax. I mistook this for the fill call int he constructor.

Reply:
That only way I found to control "__global' on method parameter [here](https://github.com/Syncleus/aparapi/pull/63/files#diff-fed28265d0be839bc56127128224a5faR639)

Reply:
@shabanovd yea your right, no worries you can ignore this comment.